src/components/User/UserTable.jsx

```diff
@@ -0,0 +1,214 @@
+        },
+        body: JSON.stringify({role : roleValue}),
+      });
+      if (!response.ok) {
+        const errorData = await response.json();
+        throw new Error(errorData.message || "Failed to update the user role");
+      }
+      
+      toast.success('User role updated successfully');
+    } catch (error) {
+      console.error(error);
+      toast.error(`Error updating user role: ${error.message}`);
+    }
+  };
+
+  /**
+   * The specified columns for the UserTable
+   */
+
+  const userTableColumns = [
+    {
+      id: "name",
+      label: "Name",
+      icon: <Bars3BottomLeftIcon />,
+      customRender: (rowData) => {
+        return <span className="mx-3">{rowData.name}</span>;
+      },
+    },
+    {
+      id: "email",
+      label: "Email",
+      icon: <AtSymbolIcon />,
+      customRender: (rowData) => {
+        return <span>{rowData.email}</span>;
+      },
+    },
+    {
+      id: "accessLevel",
+      label: "Access Level",
+      icon: <IdentificationIcon />,
+      customRender: (rowData) => {
+        return (
+          <DropdownMenu
+            submitFilters={(selectedRole) => {
+              applyRole(selectedRole)
+              rowData.role=selectedRole[0] == "Admin" ? "Admin" : "User"
+              window.location.reload();  // temporary fix: reload page whenever a role is assigned, to update role in UI
+            }}
+            label={rowData.role}
+            props={{
+              singleSelect: true,
+              filterText: "Apply Role"
+            }}
+          >
+            {consts.userAccessArray.map((roles, index) => (
+              <DropdownMenuOption
+                key={index}
+                label={roles}
+                name={roles}
+              />
+            ))}
+          </DropdownMenu>
+        );
+      },
+    },
+   {
+      id: "status",
+      label: "Status",
+      icon: <ClipboardIcon />,
+      customRender: (rowData) => {
+        return (
+          <>
+            <ToggleSwitch isActive={rowData.role !== "Inactive"} onToggle={() => handleToggle(rowData._id, rowData.role !== "Inactive" ? "Active" : "Inactive")} />
+            <div className="mx-3">
+              {rowData.role === "Inactive" ? "Inactive" : "Active"}
+            </div>
+          </>
+        );
+      },
+    },
+  ];
+  
+  return(
+    <>
+      <LoadingAnimation animated={!loading} loadText={loading} />
+      <div className="flex-grow flex-col space-y-6 mb-8">
+        <UserSearchBar setSearch={setSearchFilter} />
+
+        <Table  // Data comes in as "users" state
+          loading={loading}
+          cols={userTableColumns}
+          rows={users}
+          filter={searchFilter}
+          elementsPerPage={10}
+          onRowClick={(row, rowIndex) => {
+            setSelectedUser(row["_id"])
+            //supposed to direct to users/[id] as well, but buggy with accessing dropdown or toggle features
+          }}
+          noElements={
```

Comment:
oh, what that comment was supposed to say is that I was copying the functionality of the dog table and when you pressed a particular row, it would route you to that dog's page. So I couldn't implement the same thing for the user table, which is fine cause it wasn't a requirement of the sprint anyways, but just for future reference if we do want that functionality for users, it could be buggy